src/protocolHandler.ts

```diff
@@ -0,0 +1,60 @@
+import { URLSearchParams } from 'url';
+import * as vscode from 'vscode';
+import { CommandId } from './constants';
+import { log } from './logger';
+
+const getQuery = (uri: vscode.Uri): URLSearchParams => {
+  const query = new URLSearchParams(uri.query);
+
+  return query;
+}
+
+const openGistFactory = (paramName: 'id' | 'url') => {
+  return async (uri: vscode.Uri) => {
+    const query = getQuery(uri);
+
+    const gistParam = query.get(paramName);
+
+    if (!gistParam) {
+      log.error(`No gist ${paramName} specified in "${uri}"`);
+      throw new Error(`No gist ${paramName} specified.`);
+    }
+
+    const propName = (paramName === 'id')
+      ? 'gistId'
+      : 'gistUrl'
+
+    await vscode.commands.executeCommand(CommandId.openGist, { [propName]: gistParam });
+  }
+}
+
+export class GistPadProtocolHandler implements vscode.UriHandler {
```

Comment:
Should we add another query option to specify opening it as a workspace?

Reply:
Done.